src/io.rs

```diff
@@ -127,53 +127,70 @@ impl StreamConfigsBuilder {
     }
 
     /// set preferred sample rate
-    fn with_sample_rate(&mut self, v: f32) {
-        crate::assert_valid_sample_rate(v);
-        self.prefered.sample_rate.0 = v as u32;
+    fn with_sample_rate(&mut self, sample_rate: f32) {
+        crate::assert_valid_sample_rate(sample_rate);
+        self.prefered.sample_rate.0 = sample_rate as u32;
     }
 
-    /// buffer size
+    /// define requested hardware buffer size
     #[allow(clippy::needless_pass_by_value)]
-    fn with_latency_hint(&mut self, v: AudioContextLatencyCategory) {
-        let buffer_size: u32 = u32::try_from(RENDER_QUANTUM_SIZE).unwrap();
-        let default_buffer_size = match self.supported.buffer_size() {
-            SupportedBufferSize::Range { min, .. } => buffer_size.max(*min),
-            SupportedBufferSize::Unknown => buffer_size,
+    fn with_latency_hint(&mut self, latency_hint: AudioContextLatencyCategory) {
+        let render_quantum_size = u32::try_from(RENDER_QUANTUM_SIZE).unwrap();
+
+        // at 44100Hz sample rate (this could be even more relaxed):
```

Comment:
These changes look good to me. Indeed the previous values were quite tight, I think we could even relax the 'playback' variant later.